File: day7/sum_of_multiples.py
while True:
    try:
        user_input = input("Please enter a positive whole number: ")
        user_num = int(user_input)
        if user_num < 1:
            print("\nPlease enter a positive whole number (a number greater than zero.)\n")
            continue
        user_sum = 0
        for i in range(1, user_num):
            if i % 3 == 0 or i % 5 == 0:
                user_sum += i
        print(f"The sum of multiples of 3 or 5 below {user_input} is {user_sum}.")
        break
    except ValueError:
        print("\nPlease only enter a positive whole number.\n")

File: day7/SumOfMultiples.java
import java.util.Scanner;

public class SumOfMultiples {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("Please enter a positive whole number: ");
            String userInput = scanner.nextLine();

            try {
                int userNum = Integer.parseInt(userInput);

                if (userNum <= 1) {
                    System.out.println("\nPlease only enter a positive whole number greater than 1.\n");
                    continue;
                }

                int sum = 0;
                for (int i = 1; i < userNum; i++) {
                    if (i % 3 == 0 || i % 5 == 0) {
                        sum += i;
                    }
                }

                System.out.println("The sum of multiples of 3 or 5 below " + userNum + " is " + sum + ".");
                break;

            } catch (NumberFormatException e) {
                System.out.println("\nPlease only enter a positive whole number.\n");
            }
        }

        scanner.close();
    }
}
